### PR TITLE
Delete unused IO files

### PR DESCRIPTION
**Pull request by jamesqo**
Part of https://github.com/dotnet/coreclr/issues/4651

- I noticed that `StreamReader`, `StreamWriter`, `TextReader`, and `TextWriter` were among the types printed in [the list](https://gist.github.com/jamesqo/719d955e4e1edd78a426df1123a3c8b0/revisions), which I am pointing out because I thought their implementations lived in CoreCLR. Please tell me if I should add them back.

cc @jkotas

### CODE REVIEW

**Comment by jkotas** [comment]
You also need to do delete or refactor other code that is referencing the removed types to fix the build breaks.

> StreamReader, StreamWriter, TextReader, and TextWriter

Yes, these live in corefx. It is fine to remove them as long as everything still builds.

File: src/mscorlib/mscorlib.shared.sources.props [hunk 1]
```diff
@@ -811,39 +811,27 @@
   </ItemGroup>
   <ItemGroup>
     <IoSources Include="$(BclSourcesRoot)\System\IO\__Error.cs" />
-    <IoSources Include="$(BclSourcesRoot)\System\IO\__HResults.cs" />
```
Comment (jkotas) [inline src/mscorlib/mscorlib.shared.sources.props]: I do not think __HResults should be deleted. It is necessary to make things compile.

**Comment by jamesqo** [comment]
> You also need to do delete or refactor other code that is referencing the removed types to fix the build breaks.

Ok, sorry. I thought I had tested things locally to make sure my changes compiled, so I'm surprised that CI errored.

**Comment by jamesqo** [comment]
@jkotas I realized there are some classes that are only partially being stripped referencing the deleted types. Will remove those stripped methods as well.

**Comment by justinvp** [comment]
From your gist, it looks like you're comparing the Windows assemblies. Note that the Unix build may have slightly different dependencies due to differences in the implementation.

For example, `TimeZoneInfo` on Unix uses [`Directory.EnumerateFiles`](https://github.com/dotnet/coreclr/blob/2e67c8f8a95dddd5716e1b60705ff63c5a09a805/src/mscorlib/src/System/TimeZoneInfo.Unix.cs#L409) and [`File.ReadAllLines`](https://github.com/dotnet/coreclr/blob/2e67c8f8a95dddd5716e1b60705ff63c5a09a805/src/mscorlib/src/System/TimeZoneInfo.Unix.cs#L238) which have dependencies on `FileSystemEnumerable` and `StreamReader`.

**Comment by jamesqo** [comment]
@justinvp Ugh, that's a shame. You appear to be right, my build errored out after I passed in `osxmscorlib` and `linuxmscorlib`. I'll have to revise the script to compare the intersection of removed types/methods across all 3 platforms.

**Comment by jkotas** [comment]
> `Directory.EnumerateFiles` and `File.ReadAllLines`

To get these cleaned up eventually, we either need to refactor to use lower lever abstractions; or move the implementation from corefx to coreclr. The best strategy is case-by-case.

**Comment by jamesqo** [comment]
@jkotas Ok, I finally finished rewriting the script so it conservatively outputted types/members that were removed by BclRewriter for all 3 platforms (see [here](https://gist.github.com/jamesqo/2b848c6fec404726ac7bee1faae94324)). I have tested locally for each OS and it seems to be building. If CI is green, please re-review.

There were some files that I could not delete outright, but deleted most of:

- `Directory.cs`
- `File.cs`
- `FileSystemEnumerable.cs`

File: src/mscorlib/src/System/IO/FileSystemEnumerable.cs [hunk 2]
```diff
@@ -166,11 +134,11 @@ internal class FileSystemEnumerableIterator<TSource> : Iterator<TSource>
...
-        // empty means we know in advance that we won�t find any search results, which can happen if:
-        // 1. we don�t have a search pattern
-        // 2. we�re enumerating only the top directory and found no matches during the first call
-        // This flag allows us to return early for these cases. We can�t know this in advance for
-        // SearchOption.AllDirectories because we do a �*� search for subdirs and then use the
+        // empty means we know in advance that we won�t find any search results, which can happen if:
+        // 1. we don�t have a search pattern
+        // 2. we�re enumerating only the top directory and found no matches during the first call
+        // This flag allows us to return early for these cases. We can�t know this in advance for
+        // SearchOption.AllDirectories because we do a �*� search for subdirs and then use the
```
Comment (jamesqo) [inline src/mscorlib/src/System/IO/FileSystemEnumerable.cs]: Self-note: VSCode has been doing this whenever I open a file with weird characters-- pretty strange. I'll have to fix this later.

**Comment by jkotas** [comment]
> I open a file with weird characters-- pretty strange. I'll have to fix this later.

It is best to fix it by replacing the weird characters with plain ASCII equivalents.

LGTM otherwise. Thanks!

**Comment by jkotas** [comment]
cc @JeremyKuhne

**Comment by jkotas** [comment]
@jamesqo Thank you again!

**Pull request merged**